pkg/apis/extensions/types.go

```diff
@@ -443,10 +443,17 @@ type DaemonSetSpec struct {
 	// +optional
 	MinReadySeconds int32
 
+	// DEPRECATED.
 	// A sequence number representing a specific generation of the template.
 	// Populated by the system. It can be set only during the creation.
 	// +optional
 	TemplateGeneration int64
+
+	// The number of old history to retain to allow rollback.
+	// This is a pointer to distinguish between explicit zero and not specified.
+	// Defaults to 2.
```

Comment:
It's unlike that you will need more than three but I guess it depends on your update cadence. I would expect stuff that run as Deployments will release sooner?